packages/docs/code-snippets/external/example-image-gallery/src/js/helpers.js

```diff
@@ -0,0 +1,209 @@
+
+/**
+ * Display a message to the user in the output area.
+ * @param {string} text
+ */
+export function showMessage (text) {
+  const output = document.getElementById('output')
+  if (!output) {
+    return
+  }
+  const node = document.createElement('div')
+  node.innerText = text
+  output.appendChild(node)
+}
+
+/**
+ * Display a URL in the output area as a clickable link.
+ * @param {string} url
+ */
+export function showLink (url) {
+  const output = document.getElementById('output')
+  if (!output) {
+    return
+  }
+  const node = document.createElement('a')
+  node.href = url
+  node.target = '_external'
+  node.innerText = `> 🔗 ${url}`
+  output.appendChild(node)
+}
+
+// #endregion upload-view
+
+/**
+ * Makes a div containing an icon, followed by a text label
+ * @param {string} iconClass class for the icon, e.g. 'fontawesome-share'
+ * @param {string} labelText
+ * @returns {HTMLDivElement}
+ */
+export function iconLabel (iconClass, labelText) {
+  const label = document.createElement('span')
+  label.textContent = labelText
+  const icon = document.createElement('span')
+  icon.className = iconClass
+  const div = document.createElement('div')
+  div.appendChild(icon)
+  div.appendChild(label)
+  return div
+}
+
+/**
+ * Sets window location to the given path, if we're not already there.
+ * @param {string} path
+ */
+export function navToPath (path) {
+  if (window.location.pathname !== path) {
+    window.location.pathname = path
+  }
+}
+
+/**
+ * Set window location to the settings page.
+ */
+export function navToSettings () {
+  navToPath('/settings.html')
+}
```

Comment:
@yusefnapora this kind of navigating from JS breaks when sites are used on path-based ipfs gateways like ipfs.io. Your path will be prefixed by the root cid, like /bafy123/settings.html